src/framegrab/grabber.py

```diff
@@ -10,30 +10,43 @@
 import cv2
 import numpy as np
 
+logger = logging.getLogger(__name__)
+
 # Optional imports
 try:
     from pypylon import pylon
-except ImportError:
+except ImportError as e:
+    logger.warning(
+        "Dependency for Basler cameras (pypylon) not found. "
+        "If you are not trying to use a Basler camera, you can ignore this message. "
+        "If you are trying to use a Basler camera, you will need to install pypylon. "
+        f"{e} "
+    )
     pylon = None
```

Comment:
nit: better to log the exception object directly instead of converting it to string first - `logger.warning("message", e)` instead of `logger.warning("message {e}")` because the former can preserve the stack trace for an advanced logging system, whereas the latter throws away all information except the string message.